services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/permissions/PermissionServiceITest.kt

```diff
@@ -118,56 +121,56 @@ class PermissionServiceITest : DatabaseTest() {
         val hankemuokkaus = roleRepository.findOneByRole(Role.HANKEMUOKKAUS)
         val hakemusasiointi = roleRepository.findOneByRole(Role.HAKEMUSASIOINTI)
         val katseluoikeus = roleRepository.findOneByRole(Role.KATSELUOIKEUS)
-        val hankkeet = saveSeveralHanke(4)
+        val hankkeet = saveSeveralHanke(listOf(HANKE_TUNNUS, "HAI23-1", "HAI23-2", "HAI23-3"))
         listOf(kaikkiOikeudet, hankemuokkaus, hakemusasiointi, katseluoikeus).zip(hankkeet) {
             role,
             hanke ->
             permissionRepository.save(
                 PermissionEntity(
-                    userId = username,
+                    userId = USERNAME,
                     hankeId = hanke.id!!,
                     role = role,
                 )
             )
         }
 
-        val response = permissionService.getAllowedHankeIds(username, PermissionCode.EDIT)
+        val response = permissionService.getAllowedHankeIds(USERNAME, PermissionCode.EDIT)
 
         assertEquals(listOf(hankkeet[0].id, hankkeet[1].id), response)
     }
 
     @Test
     fun `hasPermission returns false without permissions`() {
-        assertFalse(permissionService.hasPermission(2, username, PermissionCode.EDIT))
+        assertFalse(permissionService.hasPermission(2, USERNAME, PermissionCode.EDIT))
     }
 
     @Test
     fun `hasPermission with correct permission`() {
         val kaikkiOikeudet = roleRepository.findOneByRole(Role.KAIKKI_OIKEUDET)
-        val hankeId = saveSeveralHanke(1)[0].id!!
+        val hankeId = saveSeveralHanke(listOf(HANKE_TUNNUS))[0].id!!
         permissionRepository.save(
-            PermissionEntity(userId = username, hankeId = hankeId, role = kaikkiOikeudet)
+            PermissionEntity(userId = USERNAME, hankeId = hankeId, role = kaikkiOikeudet)
         )
 
-        assertTrue(permissionService.hasPermission(hankeId, username, PermissionCode.EDIT))
+        assertTrue(permissionService.hasPermission(hankeId, USERNAME, PermissionCode.EDIT))
     }
 
     @Test
     fun `hasPermission with insufficient permissions`() {
         val hakemusasiointi = roleRepository.findOneByRole(Role.HAKEMUSASIOINTI)
-        val hankeId = saveSeveralHanke(1)[0].id!!
+        val hankeId = saveSeveralHanke(listOf(HANKE_TUNNUS))[0].id!!
         permissionRepository.save(
-            PermissionEntity(userId = username, hankeId = hankeId, role = hakemusasiointi)
+            PermissionEntity(userId = USERNAME, hankeId = hankeId, role = hakemusasiointi)
         )
 
-        assertFalse(permissionService.hasPermission(hankeId, username, PermissionCode.EDIT))
+        assertFalse(permissionService.hasPermission(hankeId, USERNAME, PermissionCode.EDIT))
     }
 
     @Test
     fun `setPermission creates a new permission`() {
-        val hankeId = saveSeveralHanke(1)[0].id!!
+        val hankeId = saveSeveralHanke(listOf(HANKE_TUNNUS))[0].id!!
```

Comment:
Feels like there's a lot of changes in this file. They could be avoided by adding a delete after the hanke creation in the failing tests.

```
permissionRepository.deleteAll()
```

Reply:
Fixed.